web/auth/auth.go

```diff
@@ -1128,6 +1130,36 @@ func passphraseRenew(c echo.Context) error {
 	return c.Redirect(http.StatusSeeOther, inst.PageURL("/auth/login", nil))
 }
 
+// Used to trade a secret for OAuth client informations
+func secretExchange(c echo.Context) error {
+	type exchange struct {
+		Secret string `json:"secret"`
+	}
+	e := new(exchange)
+
+	instance := middlewares.GetInstance(c)
+	err := json.NewDecoder(c.Request().Body).Decode(&e)
+	if err != nil {
+		return jsonapi.Errorf(http.StatusBadRequest, "%s", err)
+	}
+
+	if e.Secret == "" {
+		return jsonapi.BadRequest(errors.New("Missing secret"))
+	}
+
+	doc, err := oauth.FindClientByOnBoardingSecret(instance, e.Secret)
+
+	if err != nil {
+		return jsonapi.NotFound(err)
+	}
+
+	if doc.OnboardingSecret == "" || doc.OnboardingSecret != e.Secret {
+		return jsonapi.InvalidAttribute("secret", errors.New("Invalid secret"))
+	}
+
+	return c.JSON(http.StatusOK, doc)
```

Comment:
You should use the same format for the response that the `GET /auth/register/:client_id`: https://docs.cozy.io/en/cozy-stack/auth/#get-authregisterclient-id